src/submit_jobs.py

```diff
@@ -123,17 +132,36 @@ def read_job(job: SparkJob):
         job.repartition()
 
     # Upload extracted files to AWS S3 in Hive-partitioned Parquet
+    aws = AWSClient(logger=session.logger)
+    new_local_files = []
     for job in jobs:
-        job.upload()
+        job_upload_results = job.upload(aws)
+        new_local_files.append(job_upload_results)
+
+    # If any jobs uploaded never-seen-before files, trigger a Glue crawler
+    if any(new_local_files):
+        session.logger.info("New partitions uploaded, triggering Glue crawler")
+        aws.run_and_wait_for_crawler("ccao-data-warehouse-iasworld-crawler")
 
     # Trigger a GitHub workflow to run dbt tests once all jobs are complete
-    github = SharedGitHubSession(gh_pem_path=PATH_GH_PEM)
+    session.logger.info("All file uploads complete, triggering dbt tests")
+    github = GitHubClient(logger=session.logger, gh_pem_path=PATH_GH_PEM)
```

Comment:
Having to pass the logger into the client classes to use it feels kinda of jank. Usually I'd just use `logger = logging.getLogger(__name__)` at the top of each module, but that won't capture the Spark output :/

Reply:
**[Praise]** I honestly don't mind it! It's a little extra work when constructing clients, but it does make the logger hierarchy clear (i.e. that other services are hooking into the Spark logger).